sparse/client.go

```diff
@@ -206,22 +314,25 @@ func (client *syncClient) openServer() error {
 	}
 
 	if err != nil {
-		return fmt.Errorf("open failed, err: %s", err)
+		return errors.Wrap(err, "failed to open server")
 	}
 
 	// drain the buffer and close the body
 	_, _ = io.Copy(ioutil.Discard, resp.Body)
 	_ = resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		return fmt.Errorf("resp.StatusCode(%d) != http.StatusOK", resp.StatusCode)
+		return fmt.Errorf("failed to open server: status code=%v (%v)",
+			resp.StatusCode, http.StatusText(resp.StatusCode))
 	}
 
 	return nil
 }
 
 func (client *syncClient) closeServer() {
```

Comment:
you renamed `openServer` to `open`, so it's better to have a consistent change here like `closeServer` to `close`.

Reply:
nit: I would suggest having an interface for `syncclient` to abstract the functions to map to the exposed functions of the remote sync server like `open, close, sendHole, writeData, getChecksum, getRecordedMetadata`, and then `SyncClient` can implement the interface, so we will have a good understanding for the protocol in between and good for the future unit testing. 

```go
	router.HandleFunc("/v1-ssync/open", server.open).Methods("GET")
	router.HandleFunc("/v1-ssync/close", server.close).Methods("POST")
	router.HandleFunc("/v1-ssync/sendHole", server.sendHole).Methods("POST")
	router.HandleFunc("/v1-ssync/writeData", server.writeData).Methods("POST")
	router.HandleFunc("/v1-ssync/getChecksum", server.getChecksum).Methods("GET")
	router.HandleFunc("/v1-ssync/getRecordedMetadata", server.getRecordedMetadata).Methods("GET")

```